src/ameba/glob_utils.cr

```diff
@@ -21,6 +21,8 @@ module Ameba
     # ```
     def expand(globs)
       globs.flat_map do |glob|
+        raise ArgumentError.new("No files found matching #{glob}") if Dir[glob].empty?
+
         glob += "/**/*.cr" if File.directory?(glob)
         Dir[glob]
```

Comment:
the assertion probably needs to be moved one line below to raise on empty folders:

```suggestion
        glob += "/**/*.cr" if File.directory?(glob)
        
        raise ArgumentError.new("No files found matching #{glob}") if Dir[glob].empty?
        
        Dir[glob]
```

Reply:
Also, could you please add a test for this case?

Reply:
I originally had the assertion below `glob += "/**/*.cr" if File.directory?(glob)` however this causes the below error when running `crystal src/cli.cr`.

```
> crystal src/cli.cr                                                                                                                              
globs: ["**/*.cr"]
globs: ["lib"]
Error: No files found matching lib/**/*.cr
```

`expand` is called with `["lib"]` because it is one of the default rejected globs. But since it is empty the exception is raised.

Therefore I went down the route of only raising if the given directory doesn't exists, regardless of whether there are `.cr` files in the directory.

Interested in your thoughts on this.

Reply:
@stufro I believe now the `!lib` glob has to be added to the default globs conditionally (i.e. excluded if exists). WDYT?

Reply:
Hm yes that's a good point. I guess I was assuming that every Crystal project will have lib/ but that's not a safe assumption. 

I'll have a look at conditionally adding !lib to the default globs.

Reply:
7690074 - WDYT..? I'm not super happy with it but it's the best I could come up with